Packs/OktaASA/Integrations/OktaASA/OktaASA.py

```diff
@@ -165,18 +169,29 @@ def is_token_expired(expires_date: str) -> bool:
     return current_utc_time > expires_datetime_date
 
 
-def add_time_to_events(events: List[Dict]):
+def add_time_and_related_object_data_to_events(events: List[Dict], related_objects: Dict, add_time_mapping: bool):
     """
-    Adds the _time key to the events.
+    Adds the "_time" ,"actor", "user", "client", "project" keys values to the events.
     Args:
         events: List[Dict] - list of events to add the _time key to.
+        related_objects: Dict - A dict of events related_objects to add the related objects to.
+        add_time_mapping (bool): whether to add time mapping.
     Returns:
-        list: The events with the _time key.
+        list: The events with the _time key and related object information.
     """
+    keys_to_enhance = ["project", "server"]
     for event in events:
-        create_time = arg_to_datetime(arg=event.get("timestamp"))
-        event["_time"] = create_time.strftime(DATE_FORMAT) if create_time else None
-
+        if add_time_mapping:
+            create_time = arg_to_datetime(arg=event.get("timestamp"))
+            event["_time"] = create_time.strftime(DATE_FORMAT) if create_time else None
+        event_details = event.get("details", {})
+        for key in keys_to_enhance:
+            if key in event_details and event_details.get(key):
+                id_of_related_object = event_details.get(key)
+                related_object = related_objects.get(event_details.get(key),{})
+                if related_object.get("type") == key and related_object.get("object"):
+                    related_object_data = related_object.get("object",{})
+                    event_details[key] = related_object_data or id_of_related_object
```

Comment:
You may want to simplify this area of code as follows:
```

for key in keys_to_enhance:
    value = event_details.get(key)
    if not value:
        continue

    related_object = related_objects.get(value, {})
    if related_object.get("type") == key and related_object.get("object"):
        event_details[key] = related_object["object"] or value
```
Remove redundant .get() calls
Use continue to reduce nesting
Clean up a bit naming and structure